lib/http.c

```diff
@@ -224,7 +224,7 @@ static CURLcode h3_setup_conn(struct Curl_easy *data,
 #ifdef ENABLE_QUIC
   /* We want HTTP/3 directly, setup the filter chain ourself,
    * overriding the default behaviour. */
-  DEBUGASSERT(conn->transport = TRNSPRT_QUIC);
+  DEBUGASSERT(conn->transport == TRNSPRT_QUIC);
```

Comment:
Perhaps this can be transposed to

```c
DEBUGASSERT(TRNSPRT_QUIC == conn->transport);
```
The rationale is, you cannot accidentally assign `conn->transport` to the constant `TRNSPRT_QUIC`.

Reply:
@catull yes, that is a better way of doing it.